### PR TITLE
OOP school library: set up associations

### PR DESCRIPTION
**Pull request by arnoldnekemiah**
-     Created a class Classroom with the following:
        `@label` instance variable,  _initialized_ in the constructor.
        Setter and getter for @label (remember about attr_accessor).

-     Created the has-many/belongs-to relationship between Classroom and Student. The following should be implemented:
        Created the has-many side (a classroom has many students).
        Created the belongs-to side (a student belongs to a classroom).
        Made sure that when adding a student to a classroom it also sets the classroom for the student.
        Made sure that when setting the classroom for a student it also adds it to the classrooms' students.

-     Created a class Book with the following:

        `@title` and `@author` instance variables, should be initialized in the constructor.
        Setters and getters for instance variables (remember about attr_accessor).

-     Created a class Rental with the following:

        @date instance variable, should be initialized in the constructor.
        Setter and getter for @date (remember about attr_accessor).

-     Created the many-to-many (also has-many-through) relationship between Person and Book using the intermediate class Rental. The following should be implemented:

        Created the has-many side of Book and Rental (a book has many rentals).
        Created the belongs-to side of Rental and Book (a rental belongs to a book).
        Created the has-many side of Person and Rental (a person has many rentals).
        Created the belongs-to side of Rental and Person (a rental belongs to a person).
        Modified the constructor of Rental so Book and Person are set in it.

### CODE REVIEW

File: classroom.rb [hunk 1]
```diff
@@ -0,0 +1,16 @@
+require_relative 'student'
+
+class Classroom
+  attr_accessor :label, :students
```
Comment (mukhammadrizooff) [inline classroom.rb]: - [x] [REQUIRED] Good job so far! The `:students` should contained in `attr_reader` like:

```ruby
attr_reader :students
```

File: book.rb [hunk 2]
```diff
@@ -0,0 +1,13 @@
+class Book
+  attr_accessor :title, :author
```
Comment (mukhammadrizooff) [inline book.rb:2]: - [x] [REQUIRED] Good job so far! Getters and Setters for `@rentals` should be present in `attr_accessor` of `Book`, please add `:rentals` to that 🙏🏼 

```ruby
attr_accessor :title, :author, :rentals
```

File: student.rb [hunk 3]
```diff
@@ -1,12 +1,13 @@
-require './person'
+require_relative 'person'
 
 # This is student class, which represents student's information
 class Student < Person
-  attr_reader :classroom
+  attr_accessor :classroom
```
Comment (mukhammadrizooff) [inline student.rb]: - [x] [REQUIRED] Good job so far! The `:classroom` should contained at `attr_reader` 🙏🏼 

```ruby
attr_reader :classroom
```

**Review by mukhammadrizooff — CHANGES_REQUESTED** [review]
Hi @arnoldnekemiah, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights
- No linter errors ✔️ 
- Correct GitFlow ✔️ 
- Good implementations ✔️
- Descriptive `README` file ✔️

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by algerina — APPROVED** [review]
Hi @arnoldnekemiah , 

Great job so far!  

May I highlight the following: 
- [x]  Well donr implementing all the changes required by the previews TSE👏  ✔️ 
- [x] Your code is clean and well organized✔️



## STATUS: APPROVED :muscle::clap:

Congratulations! :tada:

Your project is complete!  It's time to merge it :shipit:


 
 ## Optional suggestions

 - Please check the comment under the review.

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._
Cheers and Happy coding! :clap: :clap: :clap:

Feel free to leave any questions or comments in the PR thread and tag me @algerina if something is not 100% clear.

#### _**Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.**_
------------------------------------------------------------------------------------------------------------------------------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using this [form.](https://airtable.com/shrQAqnBwek5a0O0s)
___
_If anything seems to be unclear to you, reach out and ask anything by commenting on this Pull Request and mention me as [Algerina](https://github.com/algerina), I will receive a notification and will respond._

[GitHub](https://github.com/algerina) | [LinkedIn](https://linkedin.com/in/amel-khiri) | [Twitter](https://twitter.com/AmelQahwadji)

**Pull request merged**